server/src/completion.ts

```diff
@@ -0,0 +1,50 @@
+import * as lsp from 'vscode-languageserver';
+
+/**
+ * Convert Angular's CompletionKind to LSP CompletionItemKind.
+ * @param kind Angular's CompletionKind
+ */
+function ngCompletionKindToLspCompletionItemKind(kind: string):
+    lsp.CompletionItemKind {
+  switch (kind) {
+    case 'attribute':
+    case 'html attribute':
+    case 'property':
+      return lsp.CompletionItemKind.Property;
+    case 'component':
+    case 'element':
+    case 'key':
+      return lsp.CompletionItemKind.Class;
+    case 'method':
+      return lsp.CompletionItemKind.Method;
+    case 'pipe':
+      return lsp.CompletionItemKind.Function;
+    case 'type':
+      return lsp.CompletionItemKind.Interface;
+    case 'reference':
+    case 'variable':
+      return lsp.CompletionItemKind.Variable;
+    case 'entity':
+    default:
+      return lsp.CompletionItemKind.Text;
+  }
+}
+
+/**
+ * Convert ts.CompletionEntry to LSP Completion Item.
+ * @param entry completion entry
+ * @param position position where completion is requested.
+ */
+export function tsCompletionEntryToLspCompletionItem(
+    entry: ts.CompletionEntry, position: lsp.Position): lsp.CompletionItem {
+  const item = lsp.CompletionItem.create(entry.name);
+  // TODO: Even though `entry.kind` is typed as ts.ScriptElementKind, it's
```

Comment:
Can we type it is a union or always assign `ts.ScriptElementKind` values to begin with? This seems dangerous.